### ISSUE TITLE
Get highlight group name of the character under the cursor

### ISSUE THREAD

**Issue by tanloong** [neovim/neovim#31318]
### Problem

Imagine in a plaintext buffer there are many words highlighted by `:call matchadd(<group>, <pattern>)`, it is hard to get the group name of the character under the cursor.

`:hlID()` does not accepts cursor position args, it asks for highlight group name as its arg and returns highlight ID. `:synID()` accepts cursor position args but it is for syntax highlight, not match highlight. The buffer is plaintext, `:Inspect` in neovim does not work either and gives `no items found at position x,y in buffer i`.

### Expected behavior

A `matID()` function for match highlight, like `synID()` for syntax highlight.

**Comment by clason** [comment]
No. If there's something `:Inspect` doesn't show, then `:Inspect` needs to be extended. (Or `matchadd` rewritten to use extmarks, if it doesn't already.)

**Comment by folke** [comment]
Since when do we have `vim.api.nvim__inspect_cell(1,0,0)`??
Never heard of that function. Great info to add to `:Inspect`
